### PR TITLE
probe-checks: add icmp check support

### PR DESCRIPTION
**Pull request by pinotelio**
This commit adds the support of ICMP checks/probes.

Signed-off-by: pinotelio [ahmadreza.mollapour@gmail.com](mailto:ahmadreza.mollapour@gmail.com)

### CODE REVIEW

**Comment by pinotelio** [comment]
PTAL: @kylegentle @liamkinne

**Comment by liamkinne** [comment]
@pinotelio Code looks good to me. Don't have the facilities to to test against the actual API. I don't have write access and likely neither does Kyle so you may have to raise a support request to get one of the Uptime guys to look at this.

**Comment by mikluko** [comment]
@pinotelio LGTM, but it requires acceptance tests coverage. Check out `*_test.go` for reference.

**Comment by pinotelio** [comment]
> @pinotelio LGTM, but it requires acceptance tests coverage. Check out `*_test.go` for reference.

@mikluko Please take a look. The test coverage and proper docs have been added.

**Comment by pinotelio** [comment]
Any update guys?
@mikluko @liamkinne @kylegentle

**Comment by kylegentle** [comment]
@pinotelio, thanks for contributing this improvement!

I haven't looked at the project in a long time, and unfortunately I don't have access to approve or merge PRs. Mikhail seems like the right person to approve.

On a separate note, it would be helpful for someone from Uptime to flesh out the Contribution Guidelines section in the README, and set expectations for how PRs will be handled.

**Comment by bazzisoft** [comment]
@pinotelio We'll do our best to look at this next week, there's a bit of a tricky situation where we're on the verge of releasing a major update v2.0 of the provider (it's been in beta for a while) which has some very significant code changes, so we'll need to check how we can port this over to that too. We'll keep you posted.

@kylegentle Thanks for your comment & good idea re the guidelines, we'll work on getting those into the README!

File: uptime/resource_uptime_check_icmp_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	"fmt"
+	"testing"
+
+	petname "github.com/dustinkirkland/golang-petname"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHeaderConversion(t *testing.T) {
```
Comment (mikluko) [inline uptime/resource_uptime_check_icmp_test.go:13]: Is this test required here or is it just a copy-paste atifact?

File: uptime/resource_uptime_check_icmp_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	hm := make(map[string]interface{})
+	hm["Foo"] = "bar"
+	hm["Baz"] = "bat"
+
+	require.Contains(t, headersMapToString(hm), "Foo: bar")
+	require.Contains(t, headersMapToString(hm), "Baz: bat")
+}
+
+func TestAccCheckHttpResource(t *testing.T) {
```
Comment (mikluko) [inline uptime/resource_uptime_check_icmp_test.go:22]: Isn't this test should be named `testAccCheckPingresource`?

**Review by mikluko — CHANGES_REQUESTED** [review]

**Comment by pinotelio** [comment]
> @pinotelio We'll do our best to look at this next week, there's a bit of a tricky situation where we're on the verge of releasing a major update v2.0 of the provider (it's been in beta for a while) which has some very significant code changes, so we'll need to check how we can port this over to that too. We'll keep you posted.
> 
> @kylegentle Thanks for your comment & good idea re the guidelines, we'll work on getting those into the README!

Since version 2.0 has released a couple of weeks ago, it's probably best to close this pull request. I'll take a look at the fresh code changes and submit a new pull request incorporating the latest updates.

**Pull request closed without merging**